src/shared/elf-util.c

```diff
@@ -860,7 +885,7 @@ int parse_elf_object(int fd, const char *executable, bool fork_disable_dump, cha
 
                 r = json_parse_file(json_in, NULL, 0, &package_metadata, NULL, NULL);
                 if (r < 0 && r != -EINVAL) /* EINVAL: json was empty, so we got nothing, but that's ok */
-                        return r;
+                        log_warning_errno(r, "Failed to read or parse json metadata, ignoring: %m");
```

Comment:
If the passed json data is truncated, `json_parse_file()` returns `-EINVAL`. Should we drop the condition `r != -EINVAL`?
Or, `if (r < 0 && package_metadata)` ??

Reply:
That's a good point. I added another commit to change json_parse_file() to return a different error in that case.